### ISSUE TITLE
% exec time incorrect when using completable future on a jetty spring web MVC controller?

### ISSUE THREAD

**Issue by HarveyEllis** [DataDog/dd-trace-java#4399]
Hi, 

I'm trying to diagnose a performance issue I've been having with a web app. 

We are using Spring MVC with Jetty in the main. dd-trace-java version 1.2.0

## Problem outline
Here is the trace:
![image](https://user-images.githubusercontent.com/23013461/206277624-1f2f1877-c314-485c-8c18-ecc1f280dfd2.png)

Key: 
- Purple is the front end web app (instrumented with RUM)
- Orange is the gateway
- Yellow is a backend service
- Other purple is postgres

The performance problem is obviously not in scope here, but the peculiar thing is the list of percentages of execution times. 
- Front end is 76.0% (looks accurate)
- Gateway is 10.1% (almost certainly not accurate)
- postgres is 9.94% (looks accurate)
- Other backend services are 0-3% each (looks accurate)

If we narrow in on the area that the backend call is occuring in, we can see that the vast majority of the time are in the purple (postgres) or the yellow (backend service):
![image](https://user-images.githubusercontent.com/23013461/206278508-fa54122c-23e9-418f-abe0-2d6677697fda.png)

Yet, the span for the gateway, which is barely in use at all, is 10% of the whole trace.

I've read [here](https://docs.datadoghq.com/tracing/glossary/#execution-time) that execution time is based on the child span at the time, or that it is based on the span that is active, [from here](https://docs.datadoghq.com/tracing/services/resource_page/#span-summary). It seems in this case that the span for the gateway is being counted as "active" or "concurrent" or something like that, even when it is not. 

## Completable future?
I was wondering whether this is because we are using completable future in the following way:
```java
@GetMapping("/my-service-path")
public CompletableFuture<ResponseEntity<MyResponseObject>> getSomeInfo() {
        return internalService.getSomeStuff()
                .thenApplyAsync(ResponseEntity::ok);
    }
```

The thing that made me think that was the fact that there is this long vertical bar between the service and the subrequests, as shown in here, which happens at the start of the request.
![Pasted image 20221207194801](https://user-images.githubusercontent.com/23013461/206281627-149b8136-e841-4ef5-a95c-5df486f14afc.png)

I'm not sure whether the is supposed to imply that the spans are happening concurrently?

Then at the end of the span there is this - the dispatcher is obviously running the `ResponseEntity::ok` completion here:
![Pasted image 20221207194903](https://user-images.githubusercontent.com/23013461/206281641-791327be-e788-4fbd-a8e1-f47439c9fcfa.png)

Additionally, the spans at the end here don't get the trace context either - they don't get runtime id and stuff like that. 

Basically, I'm just wondering whether using CompletableFuture this way is keeping this span open or active or something like that? 

I've looked around for an issue on this, but it seems that CompletableFuture should be supported like this. Found [this](https://github.com/DataDog/dd-trace-java/issues/850) from a while ago, and [this](https://github.com/DataDog/dd-trace-java/pull/2772) which mentions Jetty and completable future too. 

## Annoyances
This is really quite an annoying problem as it makes all the graphs that you try to do around tracing basically worthless (or at least very hard to interpret)

I was wondering whether anyone had experience with this?

Cheers,
Harvey

**Comment by bm1549** [comment]
@HarveyEllis  if this issue still persists on the latest version of the tracer [v1.24.2](https://github.com/DataDog/dd-trace-java/releases/tag/v1.24.2), please open up a support ticket at https://www.datadoghq.com/support/
